client/src/components/WebAuthn/Register.js

```diff
@@ -36,17 +29,29 @@ class Register extends Component {
     this.handleStartRegistration = this.handleStartRegistration.bind(this);
   }
 
-  /**
-   * keyData may not be passed through on initial render, and we can't start without it.
-   *
-   * @todo Re-evaluate whether this code is required.
-   */
-  componentWillReceiveProps() {
-    const { keyData } = this.props;
-    const { view } = this.state;
+/**
+ * React 16 deprecates componentWillReceiveProps with the new; getDerivedStateFromProps
+ * for setting state in a functional manner. This function is used by both
+ * {@see componentWillReceiveProps} and {@see getDerivedStateFromProps} in this component
+ * in order to provide the same functionality across different (supported) build versions
+ *
+ * We rely on a prop called `keyData` being present, however it may not be set upon the first
+ * render of the component. We can check and wait for it here.
+ *
+ * @param {Object} newProps incoming props for this component
+ * @param {Object} currentState current stated of this component
+ */
+  static getDerivedStateFromProps(newProps, currentState) {
+    if (currentState.view === VIEWS.LOADING && newProps.keyData) {
+      return { view: VIEWS.READY };
+    }
+    return null;
+  }
 
-    if (view === VIEWS.LOADING && keyData) {
-      this.setState({ view: VIEWS.READY });
+  componentWillReceiveProps() {
+    const stateChange = Register.getDerivedStateFromProps(this.props, this.state);
```

Comment:
I've actually just noticed that `componentWillReceiveProps()` [takes a `nextProps` parameter](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops), which represents the new props being provided. I don't know how this was working previously, but we should use that rather than `this.props` here.

Reply:
I was thinking that, but this seemed to work, so I never bothered mentioning it.

Reply:
This method is marked as unsafe in the React docs: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Reply:
Do we support React 15?

Reply:
Yeah, it needs to _work_ on React 15 so it can be used in the CMS

Reply:
Cool, so then we need both :)

Reply:
`componentDidUpdate()` exists in both and isn't deprecated, can you use that?

Reply:
Resolving this thread because its the same conversation in both now 😆  See https://github.com/silverstripe/silverstripe-webauthn-authenticator/pull/29/files/54dae60d94354b1f44e2f6c1f430baab341b845e#diff-de7962bc075fd359e3198348c4b82491